### ISSUE TITLE
Invalid URI: The hostname could not be parsed.

### ISSUE THREAD

**Issue by spyguyhi** [brandonseydel/MailChimp.Net#210]
I am getting this error when calling mailchimpManager.List.GetAllAsync() and this is the stacktrace:

   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind)
   at System.Uri..ctor(String uriString)
   at MailChimp.Net.Core.BaseLogic.CreateMailClient(String resource)
   at MailChimp.Net.Logic.ListLogic.<GetResponseAsync>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at MailChimp.Net.Logic.ListLogic.<GetAllAsync>d__3.MoveNext()

Anyone have an idea of what I might be doing wrong?  I have never come across this error and Google searching hasn't turned up anything useful.

**Comment by shadowsok** [comment]
don't instantiate the app key in the constructor instead call it separately.

MC = new MailChimpManager();
MC.Configure(new MailChimpConfiguration{ApiKey = apkey});

**Comment by spyguyhi** [comment]
Hmmm, I attempted your fix but I still got the same error.  Here's the code before:

```
MailChimp.Net.MailChimpManager mcUHFManager = new MailChimp.Net.MailChimpManager(mcUHFAPIKey);
			mcUHFManager.Configure(new MailChimp.Net.MailChimpConfiguration { Limit = 100 });
			MailChimp.Net.MailChimpManager mcAGManager = new MailChimp.Net.MailChimpManager(mcAGAPIKey);
			mcAGManager.Configure(new MailChimp.Net.MailChimpConfiguration { Limit = 100 });


			try
			{
				viewModel.mailChimpLists = mcUHFManager.Lists.GetAllAsync().Result;
			}
			catch (Exception excep)
			{
				throw;
			}
```
and here's the code after your fix:

MailChimp.Net.MailChimpManager mcUHFManager = new MailChimp.Net.MailChimpManager();
			mcUHFManager.Configure(new MailChimp.Net.MailChimpConfiguration { ApiKey = mcUHFAPIKey });
			mcUHFManager.Configure(new MailChimp.Net.MailChimpConfiguration { Limit = 100 });
			MailChimp.Net.MailChimpManager mcAGManager = new MailChimp.Net.MailChimpManager();
			mcAGManager.Configure(new MailChimp.Net.MailChimpConfiguration { ApiKey = mcAGAPIKey });
			mcAGManager.Configure(new MailChimp.Net.MailChimpConfiguration { Limit = 100 });


			try
			{
				viewModel.mailChimpLists = mcUHFManager.Lists.GetAllAsync().Result;
			}
			catch (Exception excep)
			{
				throw;
			}
MailChimp.Net.MailChimpManager mcUHFManager = new MailChimp.Net.MailChimpManager(mcUHFAPIKey);
			mcUHFManager.Configure(new MailChimp.Net.MailChimpConfiguration { Limit = 100 });
			MailChimp.Net.MailChimpManager mcAGManager = new MailChimp.Net.MailChimpManager(mcAGAPIKey);
			mcAGManager.Configure(new MailChimp.Net.MailChimpConfiguration { Limit = 100 });


			try
			{
				viewModel.mailChimpLists = mcUHFManager.Lists.GetAllAsync().Result;
			}
			catch (Exception excep)
			{
				throw;
			}
```
and here's the code after your fix:

```
MailChimp.Net.MailChimpManager mcUHFManager = new MailChimp.Net.MailChimpManager();
			mcUHFManager.Configure(new MailChimp.Net.MailChimpConfiguration { ApiKey = mcUHFAPIKey });
			mcUHFManager.Configure(new MailChimp.Net.MailChimpConfiguration { Limit = 100 });
			MailChimp.Net.MailChimpManager mcAGManager = new MailChimp.Net.MailChimpManager();
			mcAGManager.Configure(new MailChimp.Net.MailChimpConfiguration { ApiKey = mcAGAPIKey });
			mcAGManager.Configure(new MailChimp.Net.MailChimpConfiguration { Limit = 100 });


			try
			{
				viewModel.mailChimpLists = mcUHFManager.Lists.GetAllAsync().Result;
			}
			catch (Exception excep)
			{
				throw;
			}
```

both ways are caught in the try/catch with the same Invalid URI error.

**Comment by WoodsonJ** [comment]
Also having trouble with this, any further clarification would be super helpful.

Thanks!

**Comment by brandonseydel** [comment]
Please check in your branch into your public repos so we can help.

**Comment by spyguyhi** [comment]
Branch?  I haven't changed your code at all but rather added your api through nuget in our application.

**Comment by spyguyhi** [comment]
Latest news.  We were finally able to obtain VS 2017 where previously we used VS2013 so weren't able to compile the wrapper project.  I then removed the wrapper reference installed through Nuget and compiled and manually added the wrappers dll.  I no longer get this error.  I don't know if its because I downloaded the code September last year or if it is because our project is now able run the newer C# code.
